framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -129,18 +131,29 @@ ComputeMortarFunctor::operator()(const Moose::ComputeType compute_type)
     }
   };
 
-  Moose::Mortar::loopOverMortarSegments(iterators,
-                                        _assembly,
-                                        _subproblem,
-                                        _fe_problem,
-                                        _amg,
-                                        _displaced,
-                                        _mortar_constraints,
-                                        0,
-                                        _secondary_ip_sub_to_mats,
-                                        _primary_ip_sub_to_mats,
-                                        _secondary_boundary_mats,
-                                        act_functor);
+  PARALLEL_TRY
+  {
+    try
+    {
+      Moose::Mortar::loopOverMortarSegments(iterators,
+                                            _assembly,
+                                            _subproblem,
+                                            _fe_problem,
+                                            _amg,
+                                            _displaced,
+                                            _mortar_constraints,
+                                            0,
+                                            _secondary_ip_sub_to_mats,
+                                            _primary_ip_sub_to_mats,
+                                            _secondary_boundary_mats,
+                                            act_functor);
+    }
+    catch (MooseException & e)
+    {
+      _fe_problem.setException(e.what());
+    }
```

Comment:
So in `ThreadedElementLoopBase` we have a nested `try-catch` that catches `libMesh::LogicError` and `MetaPhysicL::LogicError` and re-throws as a `MooseException` which is caught by the outer `try-catch`. I think we should add that here as well. Also before setting the exception we should assert that we are not running threads (`ThreadedElementLoopBase`, because it is threaded, alternatively locks a mutex)

Reply:
That seems sane.  The double `catch(foo)` in `AuxilliarySystem::computeMortarNodalVars` has a cleaner look to it, though, for the same effect; I'll do that unless you object.

Actually ... is it sane *enough*?  Looking through our `PARALLEL_CATCH` usage elsewhere, I see `catch(MetaPhysicL::LogicError & e)` ... and for future-proofing there's no telling what we might have coming out of libraries some day.  I'd be tempted to `catch(...)` and get everything ... except that for *true* logic errors we really do want to just scream-and-die right away while there's a hope of debugging.  `parallel_object_only()` would be a perfect case in point.  It's just an artifact of past design mistakes that libMesh sees geometry errors and throws exceptions calling them logic errors, thus forcing app codes to treat `LogicError` as if it wasn't supposed to be just for bugs...

In my own code I'd used `libMesh::ConvergenceFailure` for "stuff you can catch and recover from", but that was in a world without tangled meshes, where it was just solver convergence issues forcing me to back off and retry.  I think the long run solution might be to create ... call it `libMesh::EvaluationFailure`, maybe? ... make it a subclass of `LogicError` at first for backwards compatibility, then switch the catch blocks to catch it directly, then break it out into just a subclass of `std::runtime_error`.  `ConvergenceFailure` would be a subclass of `EvaluationFailure`, too, to use for e.g. nested iterative solvers.

But that's long run.  Here I'll just make the suggested change, unless you think my immediate fears are right and we ought to be catching `MetaPhysicL::LogicError` too.

Reply:
Yes the catches at the same level look good too. And yes if you could add a catch for `MetaPhysicL::LogicError` in `AuxilliarySystem::computeMortarNodalVars` that would be much appreciated

Reply:
Plain catch and `setException(e.what())`, or `translateMetaPhysicLError`?  The latter does rethrow under the hood so we do need nested catches if you'd prefer that.